src/components/FileUploadField/index.jsx

```diff
@@ -0,0 +1,57 @@
+// We disable the rule preventing the use of ...props because these are higher order components
+/* eslint react/jsx-props-no-spreading: 0 */
+
+import React from 'react';
+import { Field } from 'formik';
+
+import { uploadFile } from 'api';
+import './style.scss';
+
+class FormikFileUpload extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filename: props.field.value,
+    };
+  }
+
+  onFileUpload = event => {
+    const { field, form, type } = this.props;
+
+    const file = event.target.files[0];
+    this.setState({ filename: file.name });
+    form.setFieldValue(field.name, file.name);
+
+    uploadFile(file, type).catch(() => {
+      alert('Failed to upload file.');
+    });
+  }
+
+  render() {
+    const { filename } = this.state;
+    const { accept } = this.props;
+
+    return (
+      <div className="FileUpload">
+        <label htmlFor="upload">
+          CHOOSE FILE
+          <input
+            id="upload"
```

Comment:
I don't like using static ids in components that can potentially be used multiple times on the same page (although it won't be in this case). I see how it's being used with `label`'s `htmlFor`, but I think both the `htmlFor` and the `id` can be removed here since the `label` is surrounding the `input`.

Reply:
I agree, and it works without the htmlFor, but also causes this linter error: `jsx-a11y/label-has-associated-control`